CAIPs/caip-211.md

```diff
@@ -0,0 +1,151 @@
+have been set, however, the meaning of any explicit values in [CAIP-25][]
+negotiations or other scope expressions can be deterministic and evolve over
+time.
+
+## Motivation
+
+Allowing experimentation and extension within certain local contexts (like a
+specific network within a namespace, or a specific community of usage of a
+network) without compromising the integrity and security of the broader
+community requires a flexible and layer mechanism for negotiating authorities
+over routing and RPC method/notification definitions. 
+
+## Specification
+
+### Implicit values
+
+Many namespaces have a single, authoritative RPC definition (whether
+human-readable, machine-readable, or both) and a set of endpoints which can be
+considered definitive. Where these can be referred to by a static URI and this
+URI is described in a [namespace profile][namespaces] of this CAIP, these static
+URIs are the "implicit" endpoint and document values for those namespaces.
+
+For a given namespace `X`, if the implicit RPC endpoints are `Y1`, `Y2` and
+`Y3`, and the implicit RPC document is `Z`, then the scope object
+
+```jsonc
+{
+  "X": {
+    methods: [A, B, C],
+    notifications: [D, E, F]
+  }
+}
+```
+
+SHOULD be interpreted as equivalent to the scope object: 
+
+```jsonc
+{
+  "X": {
+    methods: [A, B, C],
+    notifications: [D, E, F],
+    rpcEndpoints: [Y1, Y2, Y3],
+    rpcDocuments: [Z]
+  }
+}
+```
+
+### Explicit values
+
+Additional values may be set for a given `scopeObject`. 
+
+### Ordering 
+
+In the case of `rpcEndpoints`, priority is contextual and not defined
+universally. In the case of `rpcDocuments`, priority is more definitive: each
+document after the first MUST only add (and MUST NOT re-define) any terms
+already defined, iteratively through the array.
+
+For this reason, if a namespace has defined stable URIs for default
+`rpcDocuments` and `rpcEndpoints`, these SHOULD be defined in a namespace
+profile. These implicit values allow extensions and variations from those
+defaults to be negotiated, such as by [CAIP-25][] or other discovery protocols.
+
+In the context of a CAIP-25 negotiation, a requesting party may define explicit
+values for `rpcEndpoints` or `rpcDocuments` without including the implicit
+values. Extending the example above, we could say that the request:
+
+```jsonc
+{
+  "X": {
+    methods: [A, B, C],
+    notifications: [D, E, F],
+    rpcEndpoints: [U1, U2, U3],
+    rpcDocuments: [V]
+  }
+}
+```
+MUST be taken literally, as dropping the implicit values rather than appending
+to them. It MUST NOT be interpreted as equivalent to:
+
+```jsonc
+{
+  "X": {
+    methods: [A, B, C],
+    notifications: [D, E, F],
+    rpcEndpoints: [Y1, Y2, Y3, U1, U2, U3],
+    rpcDocuments: [Z, V]
+  }
+}
+```
+
+But the latter SHOULD be returned as response if dropping the implicit values in
+favor of the requested ones is unacceptable and extending the implicit values is
+preferred. 
```

Comment:
Unclear # 1: by unacceptable, I meant "allowed to extend e.g. execution APIs but I neither want to allow your documents to contradict/re-define execution APIs NOR do I want to manually confirm each connection that your documents don't contradict/redefine.  I can either make this more explicit in another commit... or change course if it's a faulty use-case assumption!

Unclear # 2: it's extraneous if it's in every request, in the sense that it verbosely states the obvious, but I wouldn't drop it from my responses if I was a provider! what a CAIP-211 profile does is give additional confidence to the providers who want to interpret no setting as equivalent to execution-apis, and let them respond without adding it.  I guess I was assuming legacy verbosity would be tolerable and just peter out over time harmlessly?

Reply:
unclear # a: I was assuming whole RPC document -- if machine-readable, "subsetting" that document or extracting, say, ABI signatures for function names can happen automatically, if human-readable (a permalink to a namespace's official dev docs, for example), this might be a little more onerous...

Reply:
> This should just be a rejected request then.

OK I'll push another commit aligned with this different behavior.